src/strings.rs

```diff
@@ -180,7 +178,8 @@ impl TypeDescriptors {
         byteorder: c_char,
         size: usize,
     ) -> Bound<'py, PyArrayDescr> {
-        let mut dtypes = self.dtypes.get(py).borrow_mut();
+        // FIXME probably a deadlock risk here due to the GIL? Might need MutexExt trait in PyO3
+        let mut dtypes = self.dtypes.lock().expect("dtype cache poisoned");
```

Comment:
Choice to use `Mutex` here makes me want to add a `MutexExt` trait to PyO3 similar to `OnceLockExt` which we already added.

Given that writes to this should be infrequent (it's a global cache, as far as I can tell), I also wonder if `RwLock` _is_ appropriate here. Readers are extremely short-lived so the analysis in https://blog.nelhage.com/post/rwlock-contention/ seems to be a non-issue (cc @alex)

... in which case I want `RwLockExt` too 😂